### PR TITLE
Some clang-tidy bugprone- fixes

### PR DESCRIPTION
**Pull request by itopcuoglu**

### CODE REVIEW

File: src/exchangeSearchData.C [hunk 1]
```diff
@@ -46,21 +46,21 @@
...
     //
-    for (i = 0; i < nsend; i++) {
+    for (int i = 0; i < nsend; i++) {
         sndPack[i].nints = sndPack[i].nreals = 0;
         sndPack[i].intData = nullptr;
         sndPack[i].realData = nullptr;
     }
     //
-    for (i = 0; i < nrecv; i++) {
+    for (int i = 0; i < nrecv; i++) {
```
[the remark this answers is not in the document]
Comment (jrood-nrel) [reply src/exchangeSearchData.C]: We should probably fix this variable shadowing. Could just use `ii` or something.

[the remark this answers is not in the document]
Comment (itopcuoglu) [reply src/exchangeSearchData.C]: Fixed this specific one, but I should keep an eye for this in general.

File: src/exchangeSearchData.C [hunk 2]
```diff
@@ -46,21 +46,21 @@
     sndPack = (PACKET*)malloc(sizeof(PACKET) * nsend);
     rcvPack = (PACKET*)malloc(sizeof(PACKET) * nrecv);
     //
-    for (i = 0; i < nsend; i++) {
+    for (int i = 0; i < nsend; i++) {
```
[the remark this answers is not in the document]
Comment (jrood-nrel) [reply src/exchangeSearchData.C:48]: I still see this one.

[the remark this answers is not in the document]
Comment (jrood-nrel) [reply src/exchangeSearchData.C:48]: We need to fix this one. Then we can merge it.

[the remark this answers is not in the document]
Comment (itopcuoglu) [reply src/exchangeSearchData.C:48]: I eliminated the redundant `int i` declaration on line 32.

**Review by jrood-nrel — APPROVED** [review]

**Pull request merged**